src/components/IconButton/index.jsx

```diff
@@ -1,28 +1,45 @@
 import React from "react";
-import useConfirmation from "hooks/useConfirmation";
+import useBooleanTimeout from "hooks/useBooleanTimeout";
 import "./IconButton.scss";
 import questionMarkIcon from "images/question-mark.png";
+import checkMarkIcon from "images/checkmark.png";
 
 const IconButton = ({
   name,
   imgSrc,
   onClickFunction,
   confirmationRequired = false,
+  giveSuccessFeedback = false,
 }) => {
   const [
     confirmationRequested,
     setConfirmationRequested,
-  ] = useConfirmation();
+  ] = useBooleanTimeout(4000);
+  const [actionSucceeded, setActionSucceeded] = useBooleanTimeout(
+    2000,
+  );
 
   const handleClick = () => {
     if (confirmationRequired && !confirmationRequested) {
       setConfirmationRequested(true);
     } else if (confirmationRequired && confirmationRequested) {
       setConfirmationRequested(false);
       onClickFunction();
+      setActionSucceeded(true);
     } else {
       onClickFunction();
+      setActionSucceeded(true);
+    }
+  };
+
+  const getIcon = () => {
+    if (confirmationRequested) {
+      return questionMarkIcon;
+    }
+    if (giveSuccessFeedback && actionSucceeded) {
+      return checkMarkIcon;
     }
+    return imgSrc;
   };
```

Comment:
I don't know that you can do this in a less complex way other than by not overloading the icon in the first place. For instance, you could redo the entire IconButton to render either a button or a confirmation or a success, and then the logic there is handled in the other components.

Put another way: right now, you're keying off of a few values in a few places and rendering an icon based on one conditional, text based on another, etc. Why not do that once, and render a new component depending on the outcome?

You know, or this is fine honestly :) I'd say `handleClick` would benefit more from tweaking than `getIcon`

Reply:
Ok cool, then I'll leave as-is and maybe revisit later if I feel like it. 😄